### PR TITLE
DAG-1219: Update evergreen.py README

### PR DESCRIPTION
**Pull request by zamj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,63 @@
...
-'MongoDB (master)'
+This is a Python client library for interacting with Evergreen and Evergreen objects. It currently only
+supports the V2 version of Evergreen's api. It can be used either by Python code in a separate application
+or on the command line to get data about Evergreen objects quickly and easily. 
+
+
+## Dependencies
+
+* Python 3.7 or later
+* Poetry 1.1 or later
```
Comment (dbradf) [inline README.md:38]: This set of dependencies is meant to be targeted at users so  the poetry required isn't applicable. That is only needed to contributors. So, we should probably move it to the `Setting up a local development environment` section.

File: README.md [hunk 2]
```diff
@@ -1,29 +1,63 @@
...
+
+## Installation
+
+```bash
+$ poetry install
```
Comment (dbradf) [inline README.md:43]: Again, this is meant for users, so we probably want to use `pip install evergreen.py` instead here.

File: README.md [hunk 3]
```diff
@@ -1,29 +1,63 @@
...
+## Table of contents
 
-You can find the documentation [here](https://evergreen-ci.github.io/evergreen.py/).
+1. [Description](#description)
+2. [Dependencies](#dependencies)
+3. [Installation](#installation)
+4. [Usage](#usage)
+5. [Documentation](#documentation)
+6. [Contributor's Guide](#contributors-guide)
+    - [High Level Architecture](#high-level-architecture)
```
Comment (dbradf) [inline README.md:16]: It looks like some of these sub-tasks missing. We should either add them, or remove them from this list.

**Review by dbradf — APPROVED** [review]

**Comment by zamj** [comment]
evergreen merge

**Pull request merged**